ext/src/CoupledSDEs.jl

```diff
@@ -0,0 +1,206 @@
+        prob = SciMLBase.remake(prob; tspan = (U(0), U(Inf)))
+    end
+    sde_function = SDEFunction(prob.f, add_noise_strength(noise_strength, prob.g, IIP))
+    prob = SciMLBase.remake(prob; f = sde_function)
+    noise_type = find_noise_type(prob, IIP)
+
+    solver, remaining = _decompose_into_sde_solver_and_remaining(diffeq)
+    integ = __init(
+        prob,
+        solver;
+        remaining...,
+        # Integrators are used exclusively iteratively. There is no reason to save anything.
+        save_start = false,
+        save_end = false,
+        save_everystep = false,
+        # DynamicalSystems.jl operates on integrators and `step!` exclusively,
+        # so there is no reason to limit the maximum time evolution
+        maxiters = Inf
+    )
+    return CoupledSDEs{IIP, D, typeof(integ), P, true}(
+        integ, deepcopy(prob.p), noise_strength, diffeq, noise_type
+    )
+end
+
+"""
+    CoupledSDEs(ds::CoupledODEs, g, p [, σ]; kwargs...)
+
+Converts a [`CoupledODEs`
+](https://juliadynamics.github.io/DynamicalSystems.jl/stable/tutorial/#CoupledODEs)
+system into a [`CoupledSDEs`](@ref).
+"""
+function DynamicalSystemsBase.CoupledSDEs(
+        ds::CoupledODEs,
+        g,
+        p, # the parameter is likely changed as the diffusion function g is added.
+        noise_strength = 1.0;
+        diffeq = DEFAULT_STOCH_DIFFEQ,
+        noise_rate_prototype = nothing,
+        noise = nothing,
+        seed = UInt64(0)
+)
+    return CoupledSDEs(
+        dynamic_rule(ds),
+        g,
+        current_state(ds),
+        p,
+        noise_strength;
+        diffeq = diffeq,
+        noise_rate_prototype = noise_rate_prototype,
+        noise = noise,
+        seed = seed
+    )
+end
+
+"""
+    CoupledODEs(ds::CoupledSDEs; kwargs...)
+
+Converts a [`CoupledSDEs`](@ref) into [`CoupledODEs`](@ref).
+"""
+function DynamicalSystemsBase.CoupledODEs(
+        sys::CoupledSDEs; diffeq = DEFAULT_DIFFEQ, t0 = 0.0)
+    return CoupledODEs(
+        dynamic_rule(sys), SVector{length(sys.integ.u)}(sys.integ.u), sys.p0;
+        diffeq = diffeq, t0 = t0
+    )
+end
+
+# Pretty print
+function DynamicalSystemsBase.additional_details(ds::CoupledSDEs)
+    solver, remaining = _decompose_into_sde_solver_and_remaining(ds.diffeq)
+    return [
+        "Noise strength" => ds.noise_strength,
+        "SDE solver" => string(nameof(typeof(solver))),
+        "SDE kwargs" => remaining,
+        "Noise type" => ds.noise_type
+    ]
+end
+
+###########################################################################################
+# API - obtaining information from the system
+###########################################################################################
+
+SciMLBase.isinplace(::CoupledSDEs{IIP}) where {IIP} = IIP
+StateSpaceSets.dimension(::CoupledSDEs{IIP, D}) where {IIP, D} = D
+DynamicalSystemsBase.current_state(ds::CoupledSDEs) = current_state(ds.integ)
+DynamicalSystemsBase.isdeterministic(ds::CoupledSDEs) = false
```

Comment:
In principle `isdeterministic` could return `true` if `ds.noise_strength = 0`

Reply:
if I understood things correctly a field called `noise_strength` won't exist in the new version

Reply:
People should make a `CoupledODEs(::CoupleSDEs)` if they want to use algorithms that need `deterministic(ds) == true`